### PR TITLE
Feature/implement make clip

### PR DESCRIPTION
**Pull request by DLorell**
***For All Submissions:***

*Proposed Changes:*

- > Implemented make_clip() and updated the test files accordingly. Fuck video files. All tests pass.

*Testing Instructions:*

* [ yes] Does simply running `pytest` suffice?

**Finally**,
* [ No] Have you updated the version number, by running `git tag v0.x`?

------

**For new features:**
* [N/A ] If you have used new packages, have you remade `requirements.txt` (explained in README)?

**For new tests:**
* [ Yes] Have you made sure your tests are in the `tests/` dir?
* [ Yes] Have you made sure your tests are in a file that starts with '`test_`'?
* [ Yes] Have you made sure your test functions starts with '`test_`'?

### CODE REVIEW

File: src/controller.py [hunk 1]
```diff
@@ -17,6 +20,43 @@ def classify_img(self, img):
...
+        #                                  outputPath: (string) "path/to/destination/video.mp4"
+
+        if isinstance(timestamp, type(None)) or isinstance(path, type(None)) \
+                                             or isinstance(timestamp[0], type(None)) \
+                                             or isinstance(timestamp[1], type(None)):
+            raise TypeError("Nonetype in arguments. Naughty naughty.")
+        if not os.path.isfile(path):
+            raise ValueError("No file at {}".format(path))
+        if isinstance(outputPath, type(None)):
+            clipPath = path[:-4] + "_subclip({},{})".format(timestamp[0], timestamp[1]) + path[-4:]
```
Comment (ralph122) [inline src/controller.py]: `os.path.splitext(path)` would be more robust here than using `path[-4:]`.

Comment (DLorell) [reply src/controller.py]: I'm not looking to get the file name, I'm extracting the file extension.

Comment (DLorell) [reply src/controller.py]: I am trying to insert text right inbetween filename and .mp4 to turn filename.mp4 -> filename_moretext.mp4

Comment (ralph122) [reply src/controller.py]: I believe that is what `splitext()` does.

Comment (DLorell) [reply src/controller.py]: I am not a smart man.

**Pull request merged**